pandas/tests/test_base.py

```diff
@@ -883,26 +883,24 @@ def get_fill_value(obj):
 
 
     def test_memory_usage(self):
+        import sys
         for o in self.objs:
             res = o.memory_usage()
-            res2 = o.memory_usage(deep=True)
+            res_deep = o.memory_usage(deep=True)
 
-            if com.is_object_dtype(o):
-                self.assertTrue(res2 > res)
+            if com.is_object_dtype(o) or (isinstance(o, Series) and com.is_object_dtype(o.index)):
+                # if there are objects, only deep will pick them up
+                self.assertTrue(res_deep > res)
             else:
-                self.assertEqual(res, res2)
+                self.assertEqual(res, res_deep)
 
             if isinstance(o, Series):
-                res = o.memory_usage(index=True)
-                res2 = o.memory_usage(index=True, deep=True)
-                if com.is_object_dtype(o) or com.is_object_dtype(o.index):
-                    self.assertTrue(res2 > res)
-                else:
-                    self.assertEqual(res, res2)
-
                 self.assertEqual(o.memory_usage(index=False) + o.index.memory_usage(),
                                  o.memory_usage(index=True))
 
+            # sys.getsizeof will call the .memory_usage defaults, and add on some GC overhead
+            self.assertAlmostEqual(res, sys.getsizeof(o), delta=100)
+
```

Comment:
I thought we disabled the test routines that one shouldn't use - in tm.TestCase
but maybe that's an open issue (or if not can you create one)

we are dropping 2.6 soon - but still like for there to be 1 way to do testing